src/language/providers/completionProvider.ts

```diff
@@ -0,0 +1,155 @@
+import { CompletionItem, CompletionItemKind, languages } from "vscode";
+import Document from "../sql/document";
+import Database from "../../database/schemas";
+import Table from "../../database/table";
+import { ObjectRef } from "../sql/types";
+
+const completionTypes = {
+  tables: {
+    order: `a`,
+    label: `table`,
+    type: `tables`,
+    icon: CompletionItemKind.File,
+  },
+  views: {
+    order: `b`,
+    label: `view`,
+    type: `views`,
+    icon: CompletionItemKind.Interface,
+  },
+  aliases: {
+    order: `c`,
+    label: `alias`,
+    type: `aliases`,
+    icon: CompletionItemKind.Reference,
+  },
+};
+
+function createCompletionItem(
+  name: string,
+  kind: CompletionItemKind,
+  detail?: string,
+  documentation?: string,
+  sortText?: string
+): CompletionItem {
+  const item = new CompletionItem(name, kind);
+  item.detail = detail;
+  item.documentation = documentation;
+  item.sortText = sortText;
+  return item;
+}
+
+async function getTableItems(schema: string, name: string) {
+  const items = await Table.getItems(schema, name);
+  return items.map((i) =>
+    createCompletionItem(i.COLUMN_NAME, CompletionItemKind.Field)
+  );
+}
+
+async function getObjectCompletions(
+  curSchema: string,
+  sqlTypes: { [index: string]: any }
+) {
+  const list = [];
+  for (let key in sqlTypes) {
+    let value = sqlTypes[key];
```

Comment:
You could cast `value` to be `sqlTypes[key] as SQLType`, but you might not need to after the interface for `completionTypes` is implemented.